src/algebra/regular.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.group
+
+/-!
+# Regular elements
+
+We introduce left-regular, right-regular and regular elements.  The final goal is to develop
+part of the API to prove, eventually, results about non-zero-divisors.
+
+-/
+variables {R : Type*}
+
+/-- A regular element is an element `c` such that multiplication by `c` both on the left and
+on the right is injective. -/
+structure is_regular [has_mul R] (c : R) : Prop :=
+(mul_left_inj : ∀ a b : R, c * a = c * b → a = b)
+(mul_right_inj : ∀ a b : R, a * c = b * c → a = b)
+
+namespace is_regular
+
+/-- A left-regular element is an element `c` such that multiplication on the left by `c`
+is injective on the left. -/
+structure is_left_regular [has_mul R] (c : R) : Prop :=
+(mul_left_inj : ∀ a b : R, c * a = c * b → a = b)
+
+/-- A right-regular element is an element `c` such that multiplication on the right by `c`
+is injective on the left. -/
+structure is_right_regular [has_mul R] (c : R) : Prop :=
+(mul_right_inj : ∀ a b : R, a * c = b * c → a = b)
```

Comment:
```suggestion
set_option old_structure_cmd true

/-- A left-regular element is an element `c` such that multiplication on the left by `c`
is injective on the left. -/
structure is_left_regular [has_mul R] (c : R) : Prop :=
(mul_left_inj : ∀ a b : R, c * a = c * b → a = b)

/-- A right-regular element is an element `c` such that multiplication on the right by `c`
is injective on the left. -/
structure is_right_regular [has_mul R] (c : R) : Prop :=
(mul_right_inj : ∀ a b : R, a * c = b * c → a = b)

/-- A regular element is an element `c` such that multiplication by `c` both on the left and
on the right is injective. -/
structure is_regular [has_mul R] (c : R) extends is_left_regular c, is_right_regular c

namespace is_regular
```
which gives you `is_regular.to_is_right_regular` for free.

Reply:
I would still prefer to have `is_regular.right` for the projection, just because it's a lot shorter.

Reply:
I am not understanding the implications of these changes, but I am happy to go with the one that seems most reasonable to you!